pkg/resources/kafka/kafka.go

```diff
@@ -238,11 +238,23 @@ func (r *Reconciler) Reconcile(log logr.Logger) error {
 		}
 	}
 
-	// Handle PDB
+	// Handle PDB for brokers
 	if r.KafkaCluster.Spec.DisruptionBudget.Create {
-		o, err := r.podDisruptionBudget(log)
+		o, err := r.podDisruptionBudgetBrokers(log)
 		if err != nil {
-			return errors.WrapIfWithDetails(err, "failed to compute podDisruptionBudget")
+			return errors.WrapIfWithDetails(err, "failed to compute podDisruptionBudget for brokers")
+		}
+		err = k8sutil.Reconcile(log, r.Client, o, r.KafkaCluster)
+		if err != nil {
+			return errors.WrapIfWithDetails(err, "failed to reconcile resource", "resource", o.GetObjectKind().GroupVersionKind())
+		}
+	}
+
+	// Handle PDB for controllers (KRaft only)
+	if r.KafkaCluster.Spec.KRaftMode && r.KafkaCluster.Spec.DisruptionBudget.Create {
```

Comment:
Maybe move this special controller PDB logic into the if above to avoid testing `r.KafkaCluster.Spec.DisruptionBudget.Create` twice.